public/components/FeatureAnywhereContextMenu/CreateAnomalyDetector/AddAnomalyDetector.tsx

```diff
@@ -0,0 +1,823 @@
+                                    size="s"
+                                  ></EuiCallOut>
+                                </EuiFlexItem>
+                              ) : null}
+
+                              {enabled ? (
+                                <EuiFlexItem>
+                                  <EuiFormRow
+                                    label="Field"
+                                    isInvalid={isInvalid(field.name, form)}
+                                    helpText={`Custom result index name must contain less than 255 characters including the prefix "opensearch-ad-plugin-result-". Valid characters are a-z, 0-9, -(hyphen) and _(underscore).`}
+                                  >
+                                    <EuiFieldText
+                                      id="resultIndex"
+                                      placeholder="Enter result index name"
+                                      prepend={CUSTOM_AD_RESULT_INDEX_PREFIX}
+                                      {...field}
+                                    />
+                                  </EuiFormRow>
+                                </EuiFlexItem>
+                              ) : null}
+                            </EuiFlexGroup>
+                          )}
+                        </Field>
+                      </MinimalAccordion>
+
+                      <MinimalAccordion
+                        id="categoricalFields"
+                        title="Categorical fields"
+                        subTitle="Split a single time series into multiple time series based on categorical fields."
+                        initialIsOpen={false}
+                        isUsingDivider={true}
+                      >
+                        <EuiText size="s">
+                          <p>
+                            The dashboard does not support high-cardinality
+                            detectors.
+                            <a
+                              href="https://opensearch.org/docs/latest/observing-your-data/ad/index/"
+                              target="_blank"
+                            >
+                              Learn more <EuiIcon type="popout" />
+                            </a>
+                          </p>
+                        </EuiText>
+                      </MinimalAccordion>
+                    </EnhancedAccordion>
+
+                    <EuiSpacer size="l" />
+                    <EuiTitle size="s">
+                      <h3>Model Features </h3>
+                    </EuiTitle>
+                    <EuiSpacer size="m" />
+
+                    <EnhancedAccordion
+                      id="modelFeatures"
+                      title="Features"
+                      isOpen={accordionsOpen.modelFeatures}
+                      onToggle={() => onAccordionToggle('modelFeatures')}
+                    >
+                      <EuiSpacer size="s" />
+
+                      <FieldArray name="featureList">
```

Comment:
I'm reusing <FeatureAccordion> part, the reason for not reusing the whole <Feature> component is that we are using the new customized accordion - `EnhancedAccordion` as the displaying style of the Feature component part

Reply:
If the only difference is the accordion, I highly suggest to refactor everything within the accordion that already exists as a reusable component. For example, everything within `FieldArray` to me looks to be identical. We could extract that out and then call it as needed in both the existing `Features` component and here. This would save us >50 lines duplicate code that we don't need to maintain twice

Reply:
We cannot reuse the whole Feature component just by refactoring the accordion because the existing Feature page is not in accordion format, only the feature list is. So I'm customizing and reusing the feature list part - `<FeatureAccordion>`. For feature anywhere, feature list is an accordion list nested in parent Feature accordion. Also, not everything within the accordion is identical, we're customizing the style of the delete and add feature buttons here as the mockup doesn't look exactly the same as the existing page

Reply:
Got it, thanks for explaining that a bit more. Ideally we should have pushed UX to keep things consistent with existing to prevent (1) code duplication and (2) UI consistency. But given the current mocks this needs to happen this way I guess.